plugins/hls-eval-plugin/src/Ide/Plugin/Eval/CodeLens.hs

```diff
@@ -275,7 +279,19 @@ runEvalCmd plId st EvalParams{..} =
 #endif
 
                 -- set up a custom log action
-#if MIN_VERSION_ghc(9,0,0)
+#if MIN_VERSION_ghc(9,2,0)
+                -- NOTE: I removed that, it was breaking the eval plugin, sometimes
```

Comment:
I spent a long time trying to adapt this chunk of code, without success, it was always failing (tests were failing saying that flush happen on a close file descriptor).

I have no idea what is the purpose of this chunk of code and tests are passing without it. It is related to logging, but why, what, is that important?

I'll try to remove it for other GHC version and see if that's pass the test too.

Do you know what was the purpose of these lines? Is that fine if I remove them for all GHC version (if tests are OK).

Reply:
I'm guessing that `withSystemTempFile` closes the file handle and then the logging hook is still on the stack, and ghc tries to flush something to it randomly and it breaks. If there is a way of popping the logging hook before `withSystemTempFile` finishes, (or before `load` fails), then I guess it won't break.

It kinda confuses me a little why the `logHandle` is closed when the `load` fails cause I assume `withSystemTempFile` will handle it.

As to the purpose, I'm guessing again that ghc will log stuff, and whatever it logs will go through the function you give it which looks like it sends them to a temporary file which the `logHandle` is for. Why send that stuff to a temp log file? No idea maybe just to look at it or even tests?

Reply:
It is a failed early attempt at capturing the stdout of the statement being run. It can safely be removed

Reply:
Thank you, I removed this bloc of code for all GHC version, as well as all the `withSystemTempFile` logic. I assumed that the log file was empty, which unfortunately leads to an empty error message.

Tests are still ok.